src/main/java/entity/Company.java

```diff
@@ -47,17 +50,45 @@ public void placeOrder(final Customer customer) {
     }
 
     private void processOrder(final Customer customer) {
-        for (Car car : cars) {
-            if (car.isFree().get()) {
-                if (!customer.isTripDone()) {
-                    car.occupy(customer);
+        LOGGER.info("The company took the order from client with ID " + customer.getId());
+        LOGGER.info("Searching for an available car for client with ID " + customer.getId() + " ...");
+
+        while (carIterator.hasNext()) {
+            Car car = carIterator.next();
+
+            if (!customer.isTripDone()) {
+
+                if (!carIterator.hasNext()) {
+                    LOGGER.info("No free cars, wait");
+                }
+
+                LOGGER.info("Car with ID " + carIterator.next().getId() + " is found for client with ID " + customer.getId());
+                car.occupy(customer);
+                carIterator.remove();
+                LOGGER.info(cars.size() + " cars are available:");
+
+                try {
+                    TimeUnit.SECONDS.sleep(1);
+                } catch (InterruptedException e) {
+                    LOGGER.error(e.getMessage());
                 }
+
+                cars.add(car);
+                car.release();
+                LOGGER.info(cars.size() + " cars are available:");
             }
         }
     }
 
-    private static List<Car> getCars(final String filePath) {
+    private static BlockingQueue<Car> getCars() {
         return new CarsParser().getCars(new CarsFileReader().parseCarList(FILE_PATH));
+    }
 
+    private void getInitialInfo() {
```

Comment:
Rename method to related functionality - displaying information about company cars.

Reply:
e.g. showCompanyParkInfo.